ls1intum/Artemis
# `Assessment`: Show text length of complaint and complaint response

### Checklist
#### General


#### Client

### Motivation and Context


As a student or tutor, it should be possible to see the allowed complaint/complaint response text limit and the current length of the request.

### Description

In order to achieve this goal, I made the following changes:

- Created a new component (`TextAreaCounterComponent`) which wraps a `<span>` containing the allowed request length and the current request length
- Counter (`<jhi-textarea-counter/>`) added only if the request was not handled or the complaint was not submitted

### Steps for Testing

Prerequisites:
- 1 Instructor/Tutor
- 1 Student
- 1 Exercise with Complaints enabled

1. Log in to Artemis
2. Assess a student submission
3. Submit a complaint as a student
4. Handle the complaint as a/an tutor/instructor
5. Verify correct functionality 

### Review Progress





#### Code Review
#### Manual Tests

### Screenshots



<details>
<summary>Screenshots</summary>

![complaint_1](https://user-images.githubusercontent.com/60005702/165521049-14a8dabf-01c1-4e16-ae5e-db0fb112c069.png)
![complaint_2](https://user-images.githubusercontent.com/60005702/165521052-2bd4edfc-5f55-481e-9ae1-c0d21d715c8f.png)
![complaint_3](https://user-images.githubusercontent.com/60005702/165521056-dc9bb5aa-a75f-4bf2-a577-7a3d88ca929e.png)
</details>

<details>
<summary>Video</summary>

https://user-images.githubusercontent.com/60005702/165520046-42cf20fc-d0be-4233-82e4-b275d90afb76.mp4
</details>
